tests/utilities/test_callables.py

```diff
@@ -297,6 +297,22 @@ def f(x: Foo):
             "type": "object",
         }
 
+    def test_function_with_pydantic_model_default(self):
+        """
+        The serialization results for a v1 and v2 schema are different so
+        this test does not attempt to compare the results. Instead it just verifies
+        that the schema is generated without error.
+        """
+        import pydantic
+
+        class Foo(pydantic.BaseModel):
+            bar: str
+
+        def f(foo: Foo = Foo(bar="baz")):
+            ...
+
+        callables.parameter_schema(f)
```

Comment:
is the resulting schema the same in both cases?

Reply:
It looks like a couple of things changed in pydantic v2 -- one is that the field `definitions` gets replaced with the standard `$defs` which we end up dropping: https://www.diffchecker.com/HuP7QiSk/

The other is that if we copy `$defs` into `definitions`, v2 generates a slightly differently formatted schema: https://www.diffchecker.com/pFFlfjDW/
 
I'm copying the `$defs` into `definitions` but there are a bunch of differences in the generated schemas.